drivers/media/i2c/ov5647.c

```diff
@@ -269,12 +299,21 @@ static int ov5647_set_virtual_channel(struct v4l2_subdev *sd, int channel)
 	return ov5647_write(sd, OV5647_REG_MIPI_CTRL14, channel_id | (channel << 6));
 }
 
+static int __sensor_init(struct v4l2_subdev *sd);
+
 static int ov5647_stream_on(struct v4l2_subdev *sd)
 {
+	struct i2c_client *client = v4l2_get_subdevdata(sd);
 	struct ov5647 *ov5647 = to_state(sd);
 	u8 val = MIPI_CTRL00_BUS_IDLE;
 	int ret;
 
+	ret = __sensor_init(sd);
```

Comment:
__sensor_init for 89 registers at 100kHz is going to take around 36ms, and hence every start streaming will be delayed by that much.
Do we delay here at the point that we've been asked to go, or at power on and each set_format to a new mode? I'm thinking shutter lag on captures and the like.

Reply:
This is a tricky one.  In normal usage, we are going to be calling S_FMT then S_STREAM straight after - meaning the delay just get shifted from one function to another and has no bearing on the capture lag.  Unless userland is doing something very clever/complicated by running s_fmt in a separate thread to some other initialisation, I don't think it makes any difference.  

Moving to s_fmt also means we need to test if the sensor is powered on, as this may not always be the case.

Reply:
S_FMT
STREAMON
STREAMOFF
STREAMON
STREAMOFF
is now slower on the second stream on. Do we care enough to worry about it?